XMonad/Hooks/WallpaperSetter.hs

```diff
@@ -102,7 +102,9 @@ instance Default WallpaperConf where
 
 -- |returns the default association list (maps name to name.jpg, non-alphanumeric characters are omitted)
 defWPNames :: [WorkspaceId] -> WallpaperList
-defWPNames xs = WallpaperList $ map (\x -> (x,WallpaperFix (filter isAlphaNum x++".jpg"))) xs
+defWPNames xs = WallpaperList $ map (\x -> (x, WallpaperFix (filter isAlphaNum x ++ "." ++ getFormat x))) xs
+    where getFormat :: [Char] -> [Char]
+          getFormat x = reverse . takeWhile isAlphaNum $ reverse x
```

Comment:
I think using something like `System.FilePath.Posix.takeExtension` here would be better

Reply:
Now that I looked at it more, I feel like we should keep the `.jpg` here. The function is useful if you want to get the wallpaper directly from the workspace name. In this version, you'd need to write the extension in your workspace name, which isn't a good UX :frowning_face: So having `.jpg` is a sensible default IMO.

One way to allow for other types, is to create another function that takes the extension as an extra parameter to the workspace list, and maybe also to the `WallpaperConf` (as `defWPNames` is called by `completeWPConf` in `wallpaperSetter`), but I'm not sure if it's worth it (I'll probably not use the module)

Reply:
Good point.  I think this is something that the documentation doesn't really explain fully; might be worth updating that part as well